rest-social-media/ballerina/response_error_interceptor.bal

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/constraint;
+
+// Handle listener errors
+service class ResponseErrorInterceptor {
+    *http:ResponseErrorInterceptor;
+
+    remote function interceptResponseError(http:RequestContext ctx, error err) 
+            returns SocialMediaBadReqeust|SocialMediaServerError {
+        ErrorDetails errorDetails = buildErrorPayload(err.message(), "");
```

Comment:
If you want the path you can obtain it from the `http:Request` object. We are exposing it as a signature parameter in the response interceptors as well.